build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/CopyRestApiTask.java

```diff
@@ -126,13 +127,15 @@ void copy() {
 
         getLogger().debug("Rest specs for project [{}] will be copied to the test resources.", projectPath);
         fileSystemOperations.copy(c -> {
+            c.setDuplicatesStrategy(DuplicatesStrategy.FAIL);
```

Comment:
These duplicate fails are not actually needed for this PR, but I realized while adding the custom duplicate check to the compat testing task, it would be good to add a duplicate check here too.

Reply:
I don't think this is necessary since `FAIL` is now the default behavior in Gradle 7.0 and later.

Reply:
ah.. it is. https://github.com/gradle/gradle/pull/15759  I will remove this.